assets/js/datasets.js

```diff
@@ -0,0 +1,195 @@
+'use strict';
+
+window.linqs = window.linqs || {};
+window.linqs.datasets = window.linqs.datasets || {};
+window.linqs.utils = window.linqs.utils || {};
+
+window.linqs.datasets.references = window.linqs.datasets.references || {};
+window.linqs.datasets.metadata = window.linqs.datasets.metadata || {};
+
+window.linqs.datasets.listAuthors = function(pub) {
+    let authors = '';
+
+    for (let i = 0; i < pub['authors'].length; i++) {
+        let text = pub['authors'][i];
+
+        authors += window.linqs.utils.listAuthor(text, i, pub['authors'], false)
+    }
+
+    return authors;
+};
+
+window.linqs.datasets.bibtex = function(pub, id) {
+    let authors = window.linqs.datasets.listAuthors(pub);
+    return window.linqs.utils.bibtex(pub, authors, id);
+};
+
+window.linqs.datasets.size = function(size) {
+    let sizes = ["B", "KB", "MB", "GB", "TB", "PB"];
+
+    let i = 0;
+    while (i < sizes.length && size >= 1024) {
+        size = size / 1024.0;
+        i++;
+    }
+
+    return size.toFixed(2).toString() + " " +sizes[i];
+};
+
+window.linqs.datasets.makeDownloadInfo = function(downloads) {
+    let downloadInfosHTML = '';
+
+    downloads.forEach(function(download, i) {
+        let link = download['href'];
+        let text = download["text"];
+        let statsHTML = "";
+
+        let sizeRaw = download['size'];
+        let md5 = download['md5'];
+
+        if (sizeRaw != null && md5 != null) {
+            let size = window.linqs.datasets.size(sizeRaw);
+
+            statsHTML = `
+                <div class='size'>
+                    (${size})
+                </div>
+                <div class='hash'>
+                    md5: ${md5}
+                </div>
+            `;
+        }
+
+        let downloadInfoHTML = `
+            <div class='link-info'>
+                <a class='download-link' href='${link}'>${text}</a>
+                ${statsHTML}
+            </div>
+        `;
+
+        downloadInfosHTML += downloadInfoHTML;
+    });
+
+    return downloadInfosHTML;
+};
+
+window.linqs.datasets.makeReference = function(referenceKey) {
+    let reference = window.linqs.datasets.references[referenceKey];
+    let authors = window.linqs.datasets.listAuthors(reference);
+    let referenceHTML = `
+        <a href='${window.linqs.utils.makePubLink(referenceKey)}'>
+            <span class='authors'>${authors}.</span>
+            <span class='title-link'>${reference['title']}.</span>
```

Comment:
These elements won't be styled because the css for this is only in pubs.css.
Just like with the JS, take a look over all the css and figure out what is common (and therefore goes in main.scss).